milkman-sio/src/main/java/milkman/plugin/sio/editor/SocketIOAspectEditor.java

```diff
@@ -0,0 +1,132 @@
+
+public class SocketIOAspectEditor implements RequestAspectEditor, ExecutionListenerAware {
+
+    private ExecutionListenerManager executionListenerManager;
+
+
+    @Override
+    public Tab getRoot(RequestContainer request) {
+        return getRoot(request, Optional.empty());
+    }
+
+    @Override
+    public Tab getRoot(RequestContainer request, Optional<ResponseContainer> existingResponse) {
+        SocketIOAspect sioAspect = request.getAspect(SocketIOAspect.class).get();
+
+        TextField textField = new JFXTextField();
+		GenericBinding<SocketIOAspect, String> binding = GenericBinding.of(
+            SocketIOAspect::getEvent, 
+            run(SocketIOAspect::setEvent)
+                .andThen(() -> sioAspect.setDirty(true)), //mark aspect as dirty propagates to the request itself and shows up in UI
+            sioAspect); 
+		textField.textProperty().bindBidirectional(binding);
+		textField.setUserData(binding); //need to add a strong reference to keep the binding from being GC-collected.
+		HBox.setHgrow(textField, Priority.ALWAYS);
+
+        ContentEditor editor = new ContentEditor();
+        editor.setEditable(true);
+        editor.setContentTypePlugins(Arrays.asList(new JsonContentType(), new PlainContentTypePlugin()));
+        setContentTypeIfPresent(editor, request);
+        editor.setContent(sioAspect::getMessage, run(sioAspect::setMessage).andThen(() -> sioAspect.setDirty(true)));
+        VBox.setVgrow(editor, Priority.ALWAYS);
+
+        var addItemBtn = new JFXButton();
+        addItemBtn.setContentDisplay(ContentDisplay.GRAPHIC_ONLY);
+        addItemBtn.setGraphic(new FontAwesomeIconView(FontAwesomeIcon.PAPER_PLANE, "1.5em"));
+        addItemBtn.getStyleClass().add("btn-add-entry");
+
+        addItemBtn.setDisable(true);
+
+        activateIfActive(sioAspect, addItemBtn, existingResponse);
+
+        executionListenerManager.listenOnExecution(request, "sio-msg-sender-listener", new ExecutionListener() {
+            @Override
+            public void onRequestStarted(RequestContainer request, ResponseContainer response) {
+            }
+
+            @Override
+            public void onRequestReady(RequestContainer request, ResponseContainer response) {
+                addItemBtn.setDisable(false);
+                addItemBtn.setOnAction(e -> response.getAspect(SocketIOResponseAspect.class).ifPresent(sio -> {
+                    if (sio.getClient().connected()) {
+                        sio.getClient().emit(sioAspect.getEvent(), sioAspect.getMessage());
+                    }
+                }));
+            }
+
+            @Override
+            public void onRequestFinished(RequestContainer request, ResponseContainer response) {
+                addItemBtn.setDisable(true);
+            }
+        });
+
+        return new Tab(
+            "Messages",
+            new VBox(new HBox(10, new Label("Event:"), textField), new Label("Message:"), editor, addItemBtn)
```

Comment:
the message and the addItemBtn should be wrapped in a stackPane to have the usual UX (similar to how it is done in the WebsocketRequestEditor)

Reply:
Maybe you could change it? I had trouble setting up even this version and if you have a clear vision how it should look, it's best if I leave it to you 🙂